File: VB/DevExpress.Calendar.Web/Tests/TestService.vb
Imports Microsoft.VisualBasic
#If DEBUG Then
	Imports System
	Imports System.Data.Services
	Imports System.Data.Services.Client
	Imports System.Linq
	Imports Microsoft.VisualStudio.TestTools.UnitTesting
	Imports DevExpress.Calendar
	Imports DevExpress.Calendar.Model
	Imports DevExpress.Data.Filtering
	Imports System.Security.Principal
Namespace DevExpress.Xpo.Services

	<TestClass> _
	Public Class TestService

		Private _session As UnitOfWork
		Private ReadOnly Property Session() As UnitOfWork
			Get
				If _session Is Nothing Then
					_session = New UnitOfWork()
					Using service As New Service()
						_session.ConnectionString = service.ConnectionString
					End Using
				End If
				Return _session
			End Get
		End Property

		<TestMethod> _
		Public Sub CreateUsers()
			Dim bHasChanges As Boolean = False

			Dim user As Resource = Session.FindObject(Of Resource)(New BinaryOperator("Email", "bill.gates@devexpress.com"))
			If user Is Nothing Then
				user = New Resource(Session)
				user.UserName = WindowsIdentity.GetCurrent().Name
				user.Name = "Bill Gates"
				user.Email = "bill.gates@devexpress.com"
				user.Save()
				bHasChanges = True
			End If

			user = Session.FindObject(Of Resource)(New BinaryOperator("Email", "bill.clinton@devexpress.com"))
			If user Is Nothing Then
				user = New Resource(Session)
				user.UserName = "DEVEXPRESS\BillC"
				user.Name = "Bill Clinton"
				user.Email = "bill.clinton@devexpress.com"
				user.Save()
				bHasChanges = True
			End If

			If bHasChanges Then
				Session.CommitChanges()
			End If
		End Sub

	End Class
End Namespace
#End If

File: CS/DevExpress.Calendar.Web/Tests/TestService.cs
#if DEBUG
namespace DevExpress.Xpo.Services {
    using System;
    using System.Data.Services;
    using System.Data.Services.Client;
    using System.Linq;
    using Microsoft.VisualStudio.TestTools.UnitTesting;
    using DevExpress.Calendar;
    using DevExpress.Calendar.Model;
    using DevExpress.Data.Filtering;
    using System.Security.Principal;

    [TestClass]
    public class TestService {
       
        UnitOfWork _session;
        UnitOfWork Session {
            get {
                if (_session == null) {
                    _session = new UnitOfWork();
                    using (Service service = new Service()) {
                        _session.ConnectionString = service.ConnectionString;
                    }
                }
                return _session;
            }
        }       

        [TestMethod]
        public void CreateUsers() {
            bool bHasChanges = false;

            Resource user = Session.FindObject<Resource>(new BinaryOperator("Email", "bill.gates@devexpress.com"));
            if (user == null) {
                user = new Resource(Session);
                user.UserName = WindowsIdentity.GetCurrent().Name;
                user.Name = "Bill Gates";
                user.Email = "bill.gates@devexpress.com";
                user.Save();
                bHasChanges = true;
            }

            user = Session.FindObject<Resource>(new BinaryOperator("Email", "bill.clinton@devexpress.com"));
            if (user == null) {
                user = new Resource(Session);
                user.UserName = @"DEVEXPRESS\BillC";
                user.Name = "Bill Clinton";
                user.Email = "bill.clinton@devexpress.com";
                user.Save();
                bHasChanges = true;
            }

            if (bHasChanges) {
                Session.CommitChanges();
            }
        }

    }
}
#endif